I added an error message to the assert, it just cannot find the temporary file. I do not know why. Maybe we can write it to somewhere else?

`java.lang.AssertionError: /tmp/junit3895615716130475389/classifier.ser does not exist`

--- a/recommender-core/src/test/java/de/unidue/ltl/recommender/core/train/RoundTripTest.java
+++ b/recommender-core/src/test/java/de/unidue/ltl/recommender/core/train/RoundTripTest.java
@@ -32,9 +32,7 @@ import org.apache.commons.io.FileUtils;
 import org.dkpro.tc.core.Constants;
 import org.junit.After;
 import org.junit.Before;
-import org.junit.Rule;
 import org.junit.Test;
-import org.junit.rules.TemporaryFolder;
 
 import com.google.gson.JsonArray;
 import com.google.gson.JsonObject;
@@ -49,24 +47,25 @@ public class RoundTripTest
     String annotationName;
     String annotationFieldName;
 
-    @Rule
-    public TemporaryFolder resultFolder = new TemporaryFolder();
-    @Rule
-    public TemporaryFolder modelLocation = new TemporaryFolder();
+    File resultFolder = null;
+    File modelLocation = null;
 
     @Before
     public void setup() throws Exception
     {
-        modelLocation.create();
-        resultFolder.create();
-        
+        File root = FileUtils.getTempDirectory();
+        resultFolder = new File(root, "resultOut/");
+        resultFolder.mkdir();
+        modelLocation = new File(root, "modelOut/");
+        modelLocation.mkdir();
+
     }
 
     @After
     public void cleanUp() throws IOException
     {
-        modelLocation.delete();
-        resultFolder.delete();
+        FileUtils.deleteQuietly(modelLocation);
+        FileUtils.deleteQuietly(resultFolder);
     }
 
     @Test
@@ -80,11 +79,10 @@ public class RoundTripTest
     private void predict() throws Exception
     {
         initPredict();
-        PredictionWithModel pwm = new PredictionWithModel(resultFolder.getRoot());
-        pwm.run(jcas, typesystem, annotationName, annotationFieldName,
-                modelLocation.getRoot());
+        PredictionWithModel pwm = new PredictionWithModel(resultFolder);
+        pwm.run(jcas, typesystem, annotationName, annotationFieldName, modelLocation);
 
-        List<File> files = getFiles(resultFolder.getRoot());
+        List<File> files = getFiles(resultFolder);
         assertEquals(1, files.size());
 
         String content = FileUtils.readFileToString(files.get(0), "utf-8");
@@ -108,8 +106,8 @@ public class RoundTripTest
 
     private void initPredict() throws IOException
     {
-        String json = FileUtils
-                .readFileToString(new File("src/test/resources/jsonPredictRequestV3small.json"), "utf-8");
+        String json = FileUtils.readFileToString(
+                new File("src/test/resources/jsonPredictRequestV3small.json"), "utf-8");
 
         JsonObject parse = new JsonParser().parse(json).getAsJsonObject();
         JsonObject document = parse.get("document").getAsJsonObject();
@@ -130,20 +128,17 @@ public class RoundTripTest
         initTrain();
         // Train Model
         TrainNewModel m = new TrainNewModel();
-        m.run(jcas, typesystem, annotationName, annotationFieldName,
-                modelLocation.getRoot());
-        assertTrue(modelLocation.getRoot().exists());
-        File theModel = new File(modelLocation.getRoot(), Constants.MODEL_CLASSIFIER);
+        m.run(jcas, typesystem, annotationName, annotationFieldName, modelLocation);
+        assertTrue(modelLocation.exists());
+        File theModel = new File(modelLocation, Constants.MODEL_CLASSIFIER);
         assertTrue(theModel.getAbsolutePath() + " does not exist", theModel.exists());
     }
 
     private void initTrain() throws IOException
     {
-//        String json = FileUtils
-//                .readFileToString(new File(System.getProperty("user.home")+"/Desktop/training.json"), "utf-8");
-        
-        String json = FileUtils
-                .readFileToString(new File("src/test/resources/jsonTrainRequestV3small.json"), StandardCharsets.UTF_8);
+        String json = FileUtils.readFileToString(
+                new File("src/test/resources/jsonTrainRequestV3small.json"),
+                StandardCharsets.UTF_8);
 
         JsonObject parse = new JsonParser().parse(json).getAsJsonObject();
         JsonObject metadata = parse.get("metadata").getAsJsonObject();

--- a/recommender-core/src/test/java/de/unidue/ltl/recommender/core/train/RoundTripTest.java
+++ b/recommender-core/src/test/java/de/unidue/ltl/recommender/core/train/RoundTripTest.java
@@ -131,7 +131,7 @@ public class RoundTripTest
         m.run(jcas, typesystem, annotationName, annotationFieldName, modelLocation);
         assertTrue(modelLocation.exists());
         File theModel = new File(modelLocation, Constants.MODEL_CLASSIFIER);
-        assertTrue(theModel.getAbsolutePath() + " does not exist", theModel.exists());
+        assertTrue(theModel.exists());
     }
 
     private void initTrain() throws IOException
@@ -153,11 +153,8 @@ public class RoundTripTest
         }
 
         jcas = casses.toArray(new String[0]);
-
         typesystem = parse.get("typeSystem").getAsString();
-
         annotationName = metadata.get("layer").getAsString();
-
         annotationFieldName = metadata.get("feature").getAsString();
     }
 }

--- a/recommender-server/src/main/java/de/unidue/ltl/recommender/server/repository/Repository.java
+++ b/recommender-server/src/main/java/de/unidue/ltl/recommender/server/repository/Repository.java
@@ -58,6 +58,9 @@ public interface Repository
      * 
      * @param irm
      *          a trained inception recommender model
+     * @param deleteSourceLocation
+     *            if the model at the source location shall be deleted after copying the folder into
+     *            the repository
      * @throws Exception
      *          in case of any error
      */

